ui/src/components/Common/MissingServiceModal.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState, useMemo } from 'react';
```

Comment:
this can be used all over to ask the user if they want to request a service before performing an action..